zrml/neo-swaps/src/math.rs

```diff
@@ -26,25 +26,36 @@ use typenum::U80;
 type Fractional = U80;
 type Fixed = FixedU128<Fractional>;
 
+const EXP_OVERFLOW_THRESHOLD: &str = "32.44892769177272";
```

Comment:
Parsing from string takes relatively long. I recommend to parse from bits:
```suggestion
// 32.44892769177272
const EXP_OVERFLOW_THRESHOLD: Fixed = Fixed::from_bits(0x20_72EC_ECDA_6EBE_EACC_40C7);
```
Test:
```rust
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
use fixed::{FixedU128, FixedI128, types::extra::{U80}};
use std::str::FromStr;

type Fixed = FixedU128<U80>;

const EXP_OVERFLOW_THRESHOLD_STRING: &str = "32.44892769177272";
const EXP_OVERFLOW_THRESHOLD: Fixed = Fixed::from_bits(0x20_72EC_ECDA_6EBE_EACC_40C7);


#[test]
fn fixed_are_the_same() {
  let from_str = Fixed::from_str(EXP_OVERFLOW_THRESHOLD_STRING).unwrap();
  assert!(from_str == EXP_OVERFLOW_THRESHOLD);
}
```

Benchmark:
```rust
#![feature(test)]

extern crate test;
use test::Bencher;

use fixed::{FixedU128, types::extra::{U80}};
use std::str::FromStr;

type Fixed = FixedU128<U80>;

const EXP_OVERFLOW_THRESHOLD_STRING: &str = "32.44892769177272";

#[bench]
fn from_string(b: &mut Bencher) {
	b.iter(|| {
	  for _i in 0..100 {  
	    let _ = test::black_box(Fixed::from_str(EXP_OVERFLOW_THRESHOLD_STRING));
    }
	});
}

#[bench]
fn from_bits(b: &mut Bencher) {
	b.iter(|| {
	  for _i in 0..100 {  
	    test::black_box(Fixed::from_bits(0x20_72EC_ECDA_6EBE_EACC_40C7));
    }
	});
}
```

Parsing from string takes 85.5ns / iter on my current machine, while parsing from bits takes about 0.45ns (190 times faster).
Obviously the gain is relatively small compared to other operations like storage io, but hey, it's free.